routes/index.js

```diff
@@ -56,16 +59,40 @@ router.post('/search', (req, res, next) => {
   var youtubeSearchPromise = getYoutubeSearch(req.headers.host, req.body.query).then(function(data){
     toReturn.youtube = JSON.parse(data); 
   });
-  promises.push(youtubeSearchPromise);
+  promises.push(youtubeSearchPromise);*/
+  
+  var twitterSearchPromise = getTwitterSearch(req.headers.host, req.body.query).then(function(data){
+	  toReturn.twitter = JSON.parse(data);
+  });
+  promises.push(twitterSearchPromise);
+  
+  var facebookSearchPromise = getFacebookSearch(req.headers.host, req.body.query).then(function(data){
+	  toReturn.facebook = JSON.parse(data);
+  });
+  promises.push(facebookSearchPromise);
 
   when.all(promises).then(function(){
+	  
+	  if (req.body.save){
+		  var url = config.mongo.url;
+		  MongoClient.connect(url,(err,db)=>{
+			  var collection = db.collection(config.mongo.collection);
+			  var time = + new Date();
+			  collection.insert({time:time, data:toReturn},(err,result)=>{
+				  if(err){
+					  console.log('Insertion error:', err);
+				  }
+				  console.log(result);
+			  });
+		  });
+	  }
       res.setHeader('Content-Type','application/json');
       res.send(JSON.stringify(toReturn)); 
   });
 
 });
 
-var getSubreddits = function(hostname, query){
+/*var getSubreddits = function(hostname, query){
```

Comment:
It's not a merge if you only trying to preserve the functionality of your code. That isn't a merge, that's a replacement.